widget/entry.go

```diff
@@ -42,6 +44,86 @@ func (e *entryRenderer) MinSize() fyne.Size {
 	return minSize.Add(fyne.NewSize(theme.Padding()*4, theme.Padding()*2))
 }
 
+// This process builds a slice of rectangles:
+// - one entry per row of text
+// - ordered by row order as they occur in multiline text
+// This process could be optimized in the scenario where the user is selecting upwards:
+// If the upwards case instead produces an order-reversed slice then only the newest rectangle would
+// require movement and resizing. The existing solution creates a new rectangle and then moves/resizes
+// all rectangles to comply with the occurance order as stated above.
+func (e *entryRenderer) buildSelection() {
+
+	e.entry.RLock()
+	curRow := e.entry.CursorRow
+	curCol := e.entry.CursorColumn
+	selRow := -1
+	selCol := -1
+	if e.entry.selecting {
+		selRow = e.entry.selectRow
+		selCol = e.entry.selectColumn
+	}
+	e.entry.RUnlock()
+
+	textRenderer := Renderer(e.text).(*textRenderer)
+
+	// Convert column, row into x,y
+	getCoordinates := func(column int, row int) (int, int) {
+		sz := textRenderer.lineSize(column, row)
+		return sz.Width + theme.Padding()*2, sz.Height*row + theme.Padding()*2
+	}
+
+	lineHeight := e.text.charMinSize().Height
+
+	// if we have a selection then we should calculate the set of boxes and add them to e.selection
+	if selRow != -1 {
```

Comment:
If we have no selection we should exit and move the remaining code to the body of the function.

Reply:
Excellent point, this is a great place to exit early from the function.